components/common/PreferencesForm.js

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react'
+import { View, Text , Picker} from 'react-native'
+import { connect } from 'react-redux'
+import PickerGenres from './PickerGenres'
+import { Header, Card, CardSection, Button, Input, Spinner } from './'
+import firebase from 'firebase'
+
+import { updatedTitle, updatedAuthor, updatedGenre, updatedPreferencesFireBase } from '../../redux/actions/preferencesActions'
+import { loginDispatch, loginDispatchFalse } from '../../redux/actions/authActions'
+
+
+const genres = [
+  {
+    label: 'Adventure',
+    value: 'Adventure',
+  },
+  {
+    label: 'Drama',
+    value: 'Drama',
+  },
+  {
+    label: 'Fiction',
+    value: 'Fiction',
+  },
+  {
+    label: 'Biography',
+    value: 'Biography',
+  },
+  {
+    label: 'Science',
+    value: 'Science',
+  },
+  {
+    label: 'Technology',
+    value: 'Technology',
+  },
+  {
+    label: 'History',
+    value: 'History',
+  },
+  {
+    label: 'Fantasy',
+    value: 'Fantasy',
+  },
+];
+
+class PreferencesForm extends Component {
+   constructor(props) {
+    super(props);
+    // this.state = {
+    //   author:'',
+    //   genre: '',
+    // };
+  }
+
+  componentWillMount(){
```

Comment:
this function repeats a lot. We should think of something so that we don't call it in each component.